File: src/problem067/solution067.hs
main = do
    content <- readFile "triangle.txt"
    let rows = [map (\n ->  read n :: Int) . words $ line | line <- lines content]
    print . maximum . foldl1 rowsum $ rows

rowsum :: [Int] -> [Int] -> [Int]
rowsum r0 r1 = map (\(a, b, c) -> a + max b c) $ zip3 r1 (0:r0) (r0 ++ [0])

File: src/problem067/solution067.py
def main():
    rows = [map(int, line.split())for line in open('triangle.txt')]
    row_sum = lambda r0, r1: [a + max(b, c) for (a, b, c) in zip(r1, r0 + [0], [0] + r0)]
    print(max(reduce(row_sum, rows)))

if __name__ == '__main__':
    main()
